### PR TITLE
Configured the Redaction type and generated using kubebuilder all the…

### PR DESCRIPTION
**Pull request by yodigos**
… files needed for the api

### CODE REVIEW

**Review by blumamir — COMMENTED** [review]
Looks good 👍 
Added few nit suggestion

File: docs/pipeline/actions/piimasking.mdx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+The "Pii Masking" Odigos Action masks span attribute values that match a blocked value list.
+
+### Use Cases
+
+- Prevent sensitive fields from accidentally leaking into traces
+- Ensure compliance with legal, privacy, or security requirements
+
+
+### Mechanism
```
Comment (blumamir) [inline docs/pipeline/actions/piimasking.mdx:14]: You can also mention that the matching is done with regular expression, which describes the mechnism that is being used

**Review by blumamir — COMMENTED** [review]
Great work, added few more comments, mainly suggestions on being more clear in the docs

File: autoscaler/controllers/actions/piimasking_controller.go [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+	if err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	for _, signal := range action.Spec.Signals {
+
+		if _, ok := piiMaskingSupportedSignals[signal]; !ok {
+
+			err = fmt.Errorf("unsupported signal: %s", signal)
+			logger.V(0).Error(nil, err.Error())
```
[the remark this answers is not in the document]
Comment (yodigos) [reply autoscaler/controllers/actions/piimasking_controller.go:58]: No no, my bad.

File: docs/pipeline/actions/attributes/piimasking.mdx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+#### Security
+
+- By default, OpenTelemetry should not record PII (Personally Identifiable Information) or sensitive data such as passwords, api tokens, etc. However, if the data is accidentally added to the telemetry signals, or if a manual instrumentation recorded it into an attribute, this action can be used to mask such data.
+
+#### Legal and Compliance
+
+- Ensure compliance with legal and privacy.
+    - EU General Data Protection Regulation (GDPR) prohibits txhe transfer of any personal data like birthdates, addresses, or ip addresses across borders without explicit consent from the data subject.
+    - Health Insurance Portability and Accountability Act (HIPAA) requires that all health data be protected from unauthorized access.
+    - Payment Card Industry (PCI) Data Security Standards prohibit logging certain things or storing them unencrypted.
```
Comment (blumamir) [inline docs/pipeline/actions/attributes/piimasking.mdx:19]: I like that we add a section regarding compliance, but suggest that we mention just the categories we are currently addressing

File: docs/pipeline/actions/attributes/piimasking.mdx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+
+### Mechanism
+
+This action processes a list of PII categories to be masked from the traces. Each PII category is linked to a regex expression that will be applied to all span attribute values. Any portion of the span attribute values that match the regex expression will be replaced with `****`, while the rest of the attribute value will remain unchanged.
```
Comment (blumamir) [inline docs/pipeline/actions/attributes/piimasking.mdx:24]: like the new text 🤩

File: docs/pipeline/actions/attributes/piimasking.mdx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+```bash
+kubectl apply -f piimasking.yaml
+```
+
+### Full Action Options
+
+The full list of options available for the "PiiMasking" action are:
+
+- `piiCategoryConfigurations` (required): An array of dictionaries where each dictionary has two keys: `mask` and `category`. `category` is the PII Category you want to mask and `mask` is a boolean value that indicates if to mask or not.
```
Comment (blumamir) [inline docs/pipeline/actions/attributes/piimasking.mdx:59]: I wonder if we should also document the supported values for `category`. How can a user discover the list value to populate here?

File: docs/pipeline/actions/introduction.mdx [hunk 6]
```diff
@@ -33,6 +33,11 @@ Actions are a powerful tool to help you manage your observability data and ensur
 
 Odigos supports a set of common actions grouped into the following categories:
 
+- [Add Cluster Info](/pipeline/actions/addclusterinfo) - Add static cluster-scoped attributes to your data.
+- [Delete Attribute](/pipeline/actions/deleteattribute) - Delete attributes by name from your data.
+- [Rename Attribute](/pipeline/actions/rename-attribute) - Rename attributes by name in your data.
+- [Probabilistic Sampler](/pipeline/actions/probabilisticsampler) - Add a random sample of your data based on a specified probability.
+- [Pii Masking](/pipeline/actions/piimasking) - Redact attributes by kind of attribute.
```
Comment (blumamir) [inline docs/pipeline/actions/introduction.mdx:40]: I think this shouldn't be here, as it is already described in [Attributes Actions](/pipeline/actions/attributes/introduction) 2 lines below. Probably merge issue

File: autoscaler/controllers/actions/piimasking_controller.go [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+	if err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	for _, signal := range action.Spec.Signals {
+
+		if _, ok := piiMaskingSupportedSignals[signal]; !ok {
+
+			err = fmt.Errorf("unsupported signal: %s", signal)
+			logger.V(0).Error(nil, "PiiMasking action invalid configration")
```
Comment (blumamir) [inline autoscaler/controllers/actions/piimasking_controller.go:58]:
```suggestion
			logger.V(0).Error(err, "PiiMasking action invalid configration")
```

File: docs/pipeline/actions/attributes/piimasking.mdx [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+#### Legal and Compliance
+
+- Ensure compliance with legal and privacy.
+    - Payment Card Industry (PCI) Data Security Standards prohibit logging certain things or storing them unencrypted.
+
+
+### Mechanism
+
+This action processes a list of PII categories to be masked from the traces. Each PII category is linked to a regex expression that will be applied to all span attribute values. Any portion of the span attribute values that match the regex expression will be replaced with `****`, while the rest of the attribute value will remain unchanged.
```
Comment (blumamir) [inline docs/pipeline/actions/attributes/piimasking.mdx:22]: `regex expression` fix

File: docs/pipeline/actions/attributes/piimasking.mdx [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+
+
+### Basic Example
+
+The basic example below shows how to configure the Pii Masking action to mask Visa and Master cards from span attribute values.
```
Comment (blumamir) [inline docs/pipeline/actions/attributes/piimasking.mdx:27]:
```suggestion
The basic example below shows how to configure the Pii Masking action to mask Visa and MasterCard credit card number from span attribute values.
```

File: docs/pipeline/actions/attributes/piimasking.mdx [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+```
+
+### Full Action Options
+
+The full list of options available for the "PiiMasking" action are:
+
+- `piiCategories` (required): An array of strings representing the PII Category you want to mask.
+
+   The available PII categories are:
+    - `CREDIT_CARD`
```
Comment (blumamir) [inline docs/pipeline/actions/attributes/piimasking.mdx:59]: We need to explain what the user can expect when setting this option

File: autoscaler/controllers/actions/piimasking_controller.go (hunk 7 above)
Comment (yodigos) [reply autoscaler/controllers/actions/piimasking_controller.go:58]: Nice catchup, I see in the probablistic it's alsi like this.. changing there too

**Review by blumamir — APPROVED** [review]

**Pull request merged**